.github/workflows/test.yml

```diff
@@ -75,9 +75,6 @@ jobs:
         working-directory: ./examples
       - name: Run Type Check
         run: pnpm run type-check
-      - name: Run Circular Dependency Check
-        run: pnpm run circular-check
```

Comment:
@himself65 adding the embedModel to the vector store base class added a lot of circular dependencies which seem to be hard to resolve.

I checked that `madge` is checking circular deps on a file basis, not on object level. Seems too strict to me - why did you add it?

Reply:
No, check circular deps on file base makes sense. It will cause incorrect behavior for bundler like using Next.js
And there is no meaning to check on object level

Reply:
Dont remove this

Reply:
This is a very common bug, see https://stackoverflow.com/a/44727578